api/handlers.go

```diff
@@ -0,0 +1,65 @@
+package api
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/SkynetLabs/scanner-queue/database"
+	accdb "github.com/SkynetLabs/skynet-accounts/database"
+	"github.com/julienschmidt/httprouter"
+	skyapi "gitlab.com/SkynetLabs/skyd/node/api"
+)
+
+func (api *API) ScanPOST(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	skylink := ps.ByName("skylink")
+	if skylink == "" {
+		skyapi.WriteError(w, skyapi.Error{"missing required parameter 'skylink'"}, http.StatusBadRequest)
+		return
+	}
+	if !accdb.ValidSkylinkHash(skylink) {
+		skyapi.WriteError(w, skyapi.Error{"invalid skylink"}, http.StatusBadRequest)
+		return
+	}
+	sl := database.Skylink{
+		Skylink:   skylink,
+		Timestamp: time.Now().UTC(),
+		Scanned:   false,
+	}
+	err := api.db.SkylinkCreate(r.Context(), sl)
+	if err != nil {
+		skyapi.WriteError(w, skyapi.Error{err.Error()}, http.StatusInternalServerError)
+		return
+	}
+	api.logger.Debugf("Added skylink %s", sl.Skylink)
+	skyapi.WriteSuccess(w)
+}
+
+func (api *API) ScanPUT(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	skylink := ps.ByName("skylink")
```

Comment:
We can refactor this into `parseSkylink` and have the `ValidSkylinkHash` check in it.

Reply:
Which part do we want to move into this new helper?
* If we put there the extraction from `ps` that won't be very helpful  on its own, as it's going to just hide which parameter we're using. That's more magical than helpful.
* if we put there the check for missing value that would hard-couple it to handlers which require that param, which is also not super helpful.

I guess the drive behind this is the fact that we have two checks - "is it empty" and "is it valid". I suggest we drop the "is it empty" one and leave it with a single check and error "invalid or missing skylink".

Otherwise the helper would look like this, which doesn't look right to me:
```go
func validateSkylink(sl string) (string, error) {
	if sl == "" {
		return "", errors.New("missing required parameter 'skylink'")
	}
	if !accdb.ValidSkylinkHash(sl) {
		return "", errors.New("invalid skylink")
	}
	return sl, nil
}
```

Reply:
After the new development where we want to store the hash of the skylink and remove the skylink itself after it's scanned, this parser method makes more sense. Adding.